client/main/dataset.js

```diff
@@ -895,6 +924,192 @@ const DataSetModel = Backbone.Model.extend({
         }
         column.levels = levels;
     },
+
+    _readTransformPB(transform, transformPB) {
+        transform.id = transformPB.id;
+        transform.name = transformPB.name;
+        transform.description = transformPB.description;
+        transform.formula = new Array(transformPB.formula.length);
+        for (let i = 0; i < transformPB.formula.length; i++)
+            transform.formula[i] = transformPB.formula[i];
+        transform.formulaMessage = new Array(transformPB.formulaMessage.length);
+        for (let i = 0; i < transformPB.formulaMessage.length; i++)
+            transform.formulaMessage[i] = transformPB.formulaMessage[i];
```

Comment:
is this copying necessary? i think the scalar js protobuf structures are just vanilla arrays that don't need to be copied (unlike python).

Reply:
(or, if you want to be conservative you can just clone them).